app/assets/javascripts/darkswarm/controllers/products_controller.js.coffee

```diff
@@ -1,41 +1,65 @@
-Darkswarm.controller "ProductsCtrl", ($scope, $filter, $rootScope, Products, OrderCycle, FilterSelectorsService, Cart, Taxons, Properties) ->
+Darkswarm.controller "ProductsCtrl", ($scope, $filter, $rootScope, Products, OrderCycle, OrderCycleResource, FilterSelectorsService, Cart, Dereferencer, Taxons, Properties, currentHub, $timeout) ->
   $scope.Products = Products
   $scope.Cart = Cart
   $scope.query = ""
   $scope.taxonSelectors = FilterSelectorsService.createSelectors()
   $scope.propertySelectors = FilterSelectorsService.createSelectors()
   $scope.filtersActive = true
-  $scope.limit = 10
+  $scope.page = 1
+  $scope.per_page = 10
   $scope.order_cycle = OrderCycle.order_cycle
-  # $scope.infiniteDisabled = true
-
-  # All of this logic basically just replicates the functionality filtering an ng-repeat
-  # except that it allows us to filter a separate list before rendering, meaning that
-  # we can get much better performance when applying filters by resetting the limit on the
-  # number of products being rendered each time a filter is changed.
-
-  $scope.$watch "Products.loading", (newValue, oldValue) ->
-    $scope.updateFilteredProducts()
-    $scope.$broadcast("loadFilterSelectors") if !newValue
-
-  $scope.incrementLimit = ->
-    if $scope.limit < Products.products.length
-      $scope.limit += 10
-      $scope.updateVisibleProducts()
-
-  $scope.$watch 'query', -> $scope.updateFilteredProducts()
-  $scope.$watchCollection 'activeTaxons', -> $scope.updateFilteredProducts()
-  $scope.$watchCollection 'activeProperties', -> $scope.updateFilteredProducts()
-
-  $scope.updateFilteredProducts = ->
-    $scope.limit = 10
-    f1 = $filter('products')(Products.products, $scope.query)
-    f2 = $filter('taxons')(f1, $scope.activeTaxons)
-    $scope.filteredProducts = $filter('properties')(f2, $scope.activeProperties)
-    $scope.updateVisibleProducts()
-
-  $scope.updateVisibleProducts = ->
-    $scope.visibleProducts = $filter('limitTo')($scope.filteredProducts, $scope.limit)
+  $scope.supplied_taxons = null
+  $scope.supplied_properties = null
+
+  $rootScope.$on "orderCycleSelected", ->
+    $scope.update_filters()
+    $scope.clearAll()
+
+  $scope.update_filters = ->
+    order_cycle_id = OrderCycle.order_cycle.order_cycle_id
+
+    return unless order_cycle_id
+
+    params = {
+      id: order_cycle_id,
+      distributor: currentHub.id
+    }
+    OrderCycleResource.taxons params, (data)=>
+      $scope.supplied_taxons = {}
+      data.map( (taxon) ->
+        $scope.supplied_taxons[taxon.id] = Taxons.taxons_by_id[taxon.id]
```

Comment:
I don't fully understand what we're doing here. We already loaded taxons in `Taxons` but request them through `OrderCycleResource`. Also, is this assignment some sort of whitelisting? I'd appreciate if you explain it a bit more as method doc in the code itself.

Reply:
I think basically there's some dereferencing magic going on, where `Taxons` contains additional data for each taxon. I'm trying to work with the existing structures here without having to rewrite mountains of Angular. :face_with_head_bandage:

Reply:
then this should be reflected as code comments IMO so the next dev is aware of it.